core/validation/initial_check.py

```diff
@@ -0,0 +1,150 @@
+        details_dict["Please select your place name"] = None
+        details_dict["A2"] = None
+
+    return details_dict
+
+
+def pre_transformation_check(submission_details: dict[str, str]) -> list[vf.ValidationFailure]:
+    """
+    Perform pre-transformation checks on the given workbook.
+
+    :param submission_details: A dictionary containing submission details.
+    :return: A list of validation failures encountered during pre-transformation checks.
+    """
+    pre_transformation_failures = []
+
+    check_form_type(submission_details, pre_transformation_failures)
+    check_round(submission_details, pre_transformation_failures)
+    check_form_input(
+        submission_details,
+        "Are you filling this in for a Town Deal or Future High Street Fund?",
+        pre_transformation_failures,
+    )
+    check_form_input(submission_details, "Please select your place name", pre_transformation_failures)
+    check_empty_sheet(submission_details, "A1", pre_transformation_failures)
+    check_empty_sheet(submission_details, "A2", pre_transformation_failures)
+
+    return pre_transformation_failures
+
+
+def check_form_type(submission_details: dict[str, str], failures: list[vf.ValidationFailure]) -> None:
+    """
+    Check if the form type is correct.
+
+    :param submission_details: A dictionary containing submission details.
+    :param failures: A list to store validation failures.
+    :return: None
+    """
+    expected_form_type = "Town Deals and Future High Streets Fund Reporting Template (v3.0)"
+    actual_form_type = submission_details["Form Version"]
+
+    if actual_form_type != expected_form_type:
+        failures.append(
+            vf.WrongFormTypeFailure(
+                expected_form_type=expected_form_type,
+                actual_form_type=actual_form_type,
+            )
+        )
+
+
+def check_round(submission_details: dict[str, str], failures: list[vf.ValidationFailure]) -> None:
+    """
+    Check if the round value is correct.
+
+    :param submission_details: A dictionary containing submission details.
+    :param failures: A list to store validation failures.
+    :return: None
+    """
+    reporting_period = submission_details["Reporting Period"]
+
+    if reporting_period not in REPORTING_PERIOD_DICT:
+        failures.append(
+            vf.InvalidReportingPeriodFailure(
+                invalid_reporting_period=reporting_period,
+                valid_reporting_periods=REPORTING_PERIOD_DICT.keys(),
+            )
+        )
+
+        return
+
+    round_value = REPORTING_PERIOD_DICT[reporting_period]
+
+    if round_value != 3:
+        failures.append(
+            vf.WrongRoundFailure(
+                expected_round=3,
+                actual_round=round_value,
+            )
+        )
+
+
+def check_form_input(submission_details: dict[str, str], question: str, failures: list[vf.ValidationFailure]) -> None:
+    """
+    Check if the form input is valid.
```

Comment:
Docstring a bit vague, could you specifically mention this function is used to check any specified field does not contain one of ["", "< Select >"] ?

I think this function could potentially be re-used in other places in future, so good to have accurate and descriptive docstring

Reply:
This is probably redundant if the above change is implemented

Reply:
In my mind, there's two types of validation we want to be able this module to do: 

1. is where we want to check if a value e.g. Fund Type is in a set of valid values e.g. HS or TD - the above mentioned `check_form_input` covers these cases
2. we also want to check that some values just aren't null e.g. Place Name - I think this is where this function comes in - I don't think we should be checking "< Select >" here because the extraction function can deal with that specific detail. This function should just be used to check if some values aren't null and raising a failure saying it wasn't entered

Reply:
Code now checks for an expected value for a given round, performing a check to see if a relevant value has been added to catch nulls as well as mismatched values.